addons/website_event_crm/tests/test_visitor_propagation.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import users
+from odoo.addons.event_crm.tests.common import TestEventCrmCommon
+
+class TestWebsiteEventCrmFlow(TestEventCrmCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.test_lang_website = cls.env['website'].create({
+            'name': 'test lang website',
+            'user_id': cls.env.ref('base.user_admin').id,
+            'language_ids': [cls.env.ref('base.lang_en').id, cls.env.ref('base.lang_fr').id]
+        })
+        cls.test_lang_visitor = cls.env['website.visitor'].sudo().create({
+            'name': 'test visitor language',
+            'lang_id': cls.env.ref('base.lang_en').id,
+            'access_token': 'f9d2ffa0427d4e4b1d740cf5eb3cdc20',
+            'website_id': cls.test_lang_website.id,
+        })
+        # 3 leads created w/ Lead Generation rules in TestEventCrmCommon: 1 per attendee and 1 per order
+        cls.test_lang_registration1, cls.test_lang_registration2 = cls.env['event.registration'].create([
+            {
+                'event_id': cls.event_0.id,
+                'visitor_id': cls.test_lang_visitor.id,
+                'email': 'test@test.example.com',
+            },
+            {
+                'event_id': cls.event_0.id,
+                'visitor_id': cls.test_lang_visitor.id,
+                'email': 'test2@test.example.com',
+            },
+        ])
+
+    @users('user_eventregistrationdesk')
+    def test_visitor_language_propagation(self):
+        leads = self.test_lang_registration1.lead_ids | self.test_lang_registration2.lead_ids
```

Comment:
Don't you have to manually call "run_on_registrations" somewhere? Maybe it's done automatically, not sure.

Reply:
It's called automatically at `event.registration` creation. I added a docstring to specify it